Refactored the entire plugin + added support for wallpaper banners

Xayer opened the pull request.

Xayer committed:
Merge pull request #14 from BenjaminMedia/development

sticky banners are not visible on mobile anymore

Xayer committed:
added some css

Xayer committed:
added NPM, Gulp and Elixir

Xayer committed:
copied WA's banner implementation, and modified them for use in this plugin

Xayer committed:
created a new bannerGroup Class & optimized the main plugin file.

The plugin file now uses the BannerGroup class instead of the hardcoded
HTML. Sticky Content Units are now under the horseshoe banners too.

Xayer committed:
Developed Banner Management logic

+ created Banner Class
+ created BannerGroup to be generate a full Horseshoe or a BannerGroup
(different size for each responsive breakpoint)
+ now you can create a horseshoe with an array, and print the HTML
code. Simple!

Xayer committed:
fixed some code in BannerGroup & started using the new structure on the rest of the plugin

Xayer committed:
added support for wallpapers

+ extracted entire admin &
+ added support for wallpaper banners
+ removed some console logs

alfhen commented on Banner.php:
```diff
@@ -0,0 +1,67 @@
+    }
+    private $bannerBreakpoints;
+
+    private $bannerTypes = [
```
Flyt denne øverst sammen med de andre class properties

alfhen commented on Banner.php:
```diff
@@ -0,0 +1,67 @@
+        return $this->code;
+    }
+
+    public function __construct($contentUnit,$breakpoint = null,$type,$sticky = false){
```
Flyt denne op til toppen af classen, lav space efter , imellem parametre

alfhen commented on Banner.php:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+
+    private function createBannerCode($cu,$breakpoint,$type,$sticky,$offset = 0){
```
Samme som construct

alfhen commented on Banner.php:
```diff
@@ -0,0 +1,67 @@
+            if($type == 'sidebanner'){
+                $stickyAttr = ($sticky)?'class="fixed text-center static" data-listen="sticky-banner"':'class="absolute text-center"';
+                $offset = 'data-offset="'.$offset.'"';
+                $banner = <<<HTML
```
Lav om til " " double quote string i stedet

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+
+use BannerGroup\BannerGroup;
+
+class bannerPlugin{
```
Classenavn med stort forbokstav

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+                wp_enqueue_script('EAS-functions', $this->getPublicFolder() . '/js/emediate-functions.js');
+            }
+            wp_enqueue_script('EAS-fif', $this->getPublicFolder() . '/js/EAS_fif.js');
+            //wp_enqueue_script('wa-manual-cu-js', $this->getPublicFolder() . '/js/wa-manual-cu.js');
```
Remove comment

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+    }
+
+    public function bannerShortcode($attrs) {
+        $a = shortcode_atts( array(
```
Undgå usigende variable navne fx. $a, $b, $c.

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+        $mobileTop = $this->getOptionOrDefault('top-mobile',$this->getOptionOrDefault('mobile-top'));
+        $wallpaperTop = $this->getOptionOrDefault('wallpaper-takeover-cu');
+
+        echo (new bannerGroup('Horseshoe Banners',[
```
smid new bannerGroup i variabel:

``` php
$bannerGroup = new bannergroup(..);
echo $bannerGroup->getHtmlCode();
```

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+        $cu = $a['cu'];
+        $offset = $a['offset'];
+
+        return (new Banner($a['cu'],$a['breakpoint'],'banner',$a['sticky'],$a['offset']));
```
Unødvendig parantes rundt om new Banner..

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+        $this->postCount++;
+        if($this->postCount >= $postsBeforeBanners){
+            if( (($this->postCount % $postsBetweenBanners++) == 0) && ($maxPostsPerPage > $this->postCount)) {
+                echo (new BannerGroup('Middle Banners',
```
Same as above..

alfhen commented on bannerGroup.php:
```diff
@@ -0,0 +1,76 @@
+    private $breakpoints;
+    private $type;
+
+    public function getHtmlCode(){
```
``` php
public static function htmlCodeFromProps($paramOne, $paramTwo, ..){
$self = new self($paramOne, $paramTwo,..);
return $self->getHtmlCode()
}
```

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+    $footerTablet = $this->getOptionOrDefault('tablet-footer');
+    $footerMobile = $this->getOptionOrDefault('mobile-footer');
+
+    $footerBannerGroup = (new BannerGroup('Middle Banners',
```
Fix this to use new function

alfhen commented on bannerPlugin.php:
```diff
@@ -0,0 +1,324 @@
+
+            <div class="panel-body">
+                ';
+                foreach($array['forms'] as $form){
```
Smid output af for loop i variabel foroven og indsæt den i stedet

Xayer committed:
refactoring stuff

+ renamed namespace
+ did loads of good practices ( I hope)
+ reorganized various properties to be above functions for readability.
